### PR TITLE
Rename `--query` to `--filter`

### LINKED ISSUE #273: [Feature Request] Change `--query` to `--filter`
### Is your feature request related to a problem? Please describe.

`--query` might be confusing with Queries

https://docs.temporal.io/tctl/workflow/list/#--query

### Describe the solution you'd like

Change to `--filter`, since the argument is a List Filter

cc @flossypurse

### PR DESCRIPTION
**Pull request by lorensr**
Fixes #273

### CODE REVIEW

File: cli/flags.go [hunk 1]
```diff
@@ -95,9 +95,8 @@ var (
 	FlagResetReapplyType           = "reset-reapply-type"
 	FlagResetPointsOnly            = "reset-points-only"
 	FlagResetBadBinaryChecksum     = "reset-bad-binary-checksum"
-	FlagQuery                      = "query"
-	FlagQueryAlias                 = []string{"q"}
```
Comment (lorensr) [inline cli/flags.go:99]: "f" alias conflicts with follow, so I didn't include. However, `FlagFollowAlias` is only used in `workflow show`, and `FlagFilter` isn't, so I guess we could have multiple "f" aliases?

**Comment by feedmeapples** [comment]
Discussed with @bergundy, @alexshtin and @flossypurse 

As `--query` may also accept ORDER BY, renaming to `--filter` does not really communicate it well. Decided to leave as `--query`

**Comment by lorensr** [comment]
Gotcha, makes sense. Wonder if we should rename List Filter then?

**Comment by feedmeapples** [comment]
> Gotcha, makes sense. Wonder if we should rename List Filter then?

let's proceed with renaming to avoid confusion with https://docs.temporal.io/application-development/features/#queries
cc @flossypurse

**Review by feedmeapples — APPROVED** [review]

**Review by flossypurse — APPROVED** [review]
Cool!

**Comment by feedmeapples** [comment]
as per additional discussions we again may want to keep `--query` as filter still may not communicate it the best. We may have a query builder in SDKs in future in which case we can add `--filter` as a Slice Flag and `--order-by`

**Pull request closed without merging**